acme_tiny.py

```diff
@@ -1,20 +1,19 @@
 #!/usr/bin/env python
 # Copyright Daniel Roesler, under MIT license, see LICENSE at github.com/diafygi/acme-tiny
-import argparse, subprocess, json, os, sys, base64, binascii, time, hashlib, re, copy, textwrap, logging
-try:
-    from urllib.request import urlopen, Request # Python 3
-except ImportError:
-    from urllib2 import urlopen, Request # Python 2
+import argparse, subprocess, json, os, sys, base64, binascii, time, hashlib, re, textwrap, logging
+if sys.version_info[0] >= 3:
```

Comment:
much better than try...except

Reply:
This is a bit of an anti-pattern, as you shouldn't rely on the versioning information to predict the standard library API. That said, the existing solution has its own issues. A better option would be using `future.standard_library.hooks`, which would also remove the redundancies.

Reply:
Wouldn't that require installing a non-stdlib module? https://pypi.org/project/future/

It isn't much of an anti-pattern if at all, though it is a bit better as far as self-documenting goes.

Reply:
For me, I would remove Python2 support at all.

btw, should I revert this change ?

Reply:
Ack, crud, yeah.

I agree that it's more self-documenting, but it also relies on an assumed association between the value of `sys.version_info` and the layout of the API. You can rely on that in current versions of CPython, but there's no guarantee of that being the case in other runtimes or in the future.

Reply:
Ok, I will revert this chunk. I'm just curious. I'm not a native English speaker. How would you explain "Ack, crud, yeah." ? Google tells me something weird.

Reply:
> For me, I would remove Python2 support at all.

Me too. Would save a couple lines. :-)

Reply:
"Ack, crud, yeah." = Oops. Yes, you're correct.

Unfortunately, Python 2 support needs to remain in place for now, as it is still the default on some OSes.